Automattic/wp-calypso
# Theme Showcase: New component SearchThemes

#### Proposed Changes

This PR implements a new theme search experience as defined in 0EogtxvlmxMGcnEEaMVECv-fi-1259%3A37689.
![Screen Shot 2022-11-09 at 4 36 17 PM](https://user-images.githubusercontent.com/797888/200780532-80306c9a-4d69-4637-9dac-9be74af13ef7.png)

To accomplish this, we have:
1. Created a new component `<SearchThemes/>` and,
2. Updated the existing component `<KeyedSuggestions/>` to support new UX behaviors.

Note that the `<SearchThemes />` component is based on the existing component `<ThemesMagicSearchCard />` ([src](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/themes/themes-magic-search-card/index.jsx)), and for the most part they should behave similar when it comes to text injection based on suggestion selection. 

The new search experience is as follows:

Whenever the search input field is clicked, the suggestion list will show up and display three main filters: "Feature", "Columns" and "Subject".
![Screen Shot 2022-11-09 at 6 35 33 PM](https://user-images.githubusercontent.com/797888/200808001-b05fcae4-5630-4935-a644-8dbc1cc102a2.png)

Clicking on any of the above will refine the suggestion list to the list of terms from the selected filters. For instance, selecting "Subject" will inject the text `subject:` and update the suggestion list to terms under the filter "Subject", such as "Blog", "Business", etc.
![Screen Shot 2022-11-09 at 6 35 51 PM](https://user-images.githubusercontent.com/797888/200808060-65f3ee3d-0ce6-4903-aa4a-27685fcf1c6c.png)

Clicking on a term will trigger search which updates the theme grid below the search bar. For instance, selecting "Blog" will update the text to `subject:blog` and close the suggestion list. 
![Screen Shot 2022-11-09 at 6 36 19 PM](https://user-images.githubusercontent.com/797888/200808164-d0581a98-f1d6-4410-88e8-ba61b539cae0.png)

Unlike the old search experience, the suggestion list will no longer auto-suggest the first result. This is to allow users to do a plain-text search without being "forced" to select one of the suggestion. To understand more about this, try typing "blog" in the old search experience and do a plain-text search. Note that in the new search experience, keyboard navigation is still available.
![Screen Shot 2022-11-09 at 6 36 44 PM](https://user-images.githubusercontent.com/797888/200808267-bfaf56d2-ac3f-4633-abb6-ef0f2c66775b.png)

#### Testing Instructions



* Head to the theme showcase `/themes/${site_slug}`.
  * If using calypso.live, the flag `themes/showcase-i4/search-and-filter` is required.
* Ensure that the search resultset is the same as `wordpress.com/themes/${site_slug}`.
* Ensure that the search and suggestion behavior corresponds to the one described above.
* Ensure that the old search component works as it used to.
  * Disable the new component via flag, i.e.: `/themes/${site_slug}?flags=-themes/showcase-i4/search-and-filter`

#### Pre-merge Checklist






Related to #69658
